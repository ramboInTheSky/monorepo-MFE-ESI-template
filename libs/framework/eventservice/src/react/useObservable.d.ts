import { ESB } from "../esb";
export declare const useCommonObservable: (esb: ESB, callback: (data: any) => void) => void;
