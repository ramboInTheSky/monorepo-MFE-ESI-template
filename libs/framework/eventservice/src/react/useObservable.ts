import {useEffect} from "react"
import {ESB} from "../esb"

export const useCommonObservable = (esb: ESB, callback: (data: any) => void) => {
    return useEffect(() => {
        const {subscription} = esb.subscribe(callback)
        return () => {
            subscription.unsubscribe()
        }
    // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [])
}
