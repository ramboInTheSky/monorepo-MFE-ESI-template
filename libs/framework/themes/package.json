{
  "name": "@monorepo/themes",
  "version": "17.0.0",
  "author": "",
  "description": "UI theming",
  "main": "./lib/index.js",
  "module": "./lib/index.es.js",
  "types": "./lib/index.d.ts",
  "files": [
    "lib/**/*"
  ],
  "scripts": {
    "build-lib": "rollup -c && yalc push --safe --replace",
    "build-ci": "rollup -c rollup.config.prod.js",
    "lint": "eslint --ext .tsx,.ts src/",
    "tsc": "tsc",
    "test": "jest src/"
  },
  "devDependencies": {
    "@monorepo/eslint": "^3.3.0",
    "@types/jest": "24.9.1",
    "@types/node": "12.12.28",
    "@typescript-eslint/parser": "2.19.2",
    "jest": "24.9.0",
    "jest-coverage-ratchet": "0.2.3",
    "jest-junit": "10.0.0",
    "ts-jest": "24.3.0",
    "typescript": "3.8.2",
    "yalc": "^1.0.0-pre.53"
  },
  "dependencies": {
    "@mui/material": "^5.2.8",
    "@mui/styles": "^5.2.3",
    "@monorepo/core-logger": "^7.3.0"
  },
  "peerDependencies": {
    "@mui/material": "^5.2.8",
    "@mui/styles": "^5.2.3",
    "@monorepo/core-logger": "^7.3.0"
  },
  "gitHead": "",
  "jest": {
    "preset": "ts-jest"
  }
}
