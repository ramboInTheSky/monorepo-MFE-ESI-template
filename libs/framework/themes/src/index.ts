export * from "./themecolors"
export * from "./fonts"
export * from "./breakpoints"
export * from "./themeprovider"
export * from "./isThemeValid"
export * from "./theme"
export * from "../__mocks__/mockTheme"
