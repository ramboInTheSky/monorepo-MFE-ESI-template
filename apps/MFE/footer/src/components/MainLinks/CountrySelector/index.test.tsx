import React from "react"
import {render} from "@testing-library/react"
import {SCThemeProvider} from "@monorepo/theme-provider"
import {mockTheme} from "@monorepo/themes"
import {CountrySelector} from "."

const props = {
    textColor: "#000",
    siteUrl: "http://superman.com",
    elements: [
        {
            type: "CountryLink",
            url: "",
            openInNewWindow: false,
            name: "CountryLink",
            icon: "",
            text: "Saudi Arabia",
            accessibilityText: "Saudi Arabia",
            tooltip: "",
            accessibilityTooltip: "",
            description: "",
            accessibilityDescription: "",
        },
        {
            type: "CountryLanguage",
            url: "//www.amido.com/sa/ar",
            openInNewWindow: false,
            name: "CountryLanguage",
            icon: "",
            text: "AR",
            accessibilityText: "Change to Saudi Arabia in Arabic",
            tooltip: "",
            accessibilityTooltip: "",
            description: "",
            accessibilityDescription: "",
        },
        {
            type: "CountryLanguage",
            url: "//www.amido.com/sa/en",
            openInNewWindow: false,
            name: "CountryLanguage",
            icon: "",
            text: "EN",
            accessibilityText: "Change to Saudi Arabia in English",
            tooltip: "",
            accessibilityTooltip: "",
            description: "",
            accessibilityDescription: "",
        },
        {
            type: "CountryLink",
            url: "",
            openInNewWindow: false,
            name: "CountryLink",
            icon: "",
            text: "United Arab Emirates",
            accessibilityText: "United Arab Emirates",
            tooltip: "",
            accessibilityTooltip: "",
            description: "",
            accessibilityDescription: "",
        },
        {
            type: "CountryLanguage",
            url: "//www.amido.com/ae/ar",
            openInNewWindow: false,
            name: "CountryLanguage",
            icon: "",
            text: "AR",
            accessibilityText: "Change to United Arab Emirates in Arabic",
            tooltip: "",
            accessibilityTooltip: "",
            description: "",
            accessibilityDescription: "",
        },
    ],
}

describe("Components - MainLinks/CountrySelector: ", () => {
    it("should match the snapshot template", () => {
        const {asFragment, queryByTestId} = render(
            <SCThemeProvider theme={mockTheme}>
                <CountrySelector {...props} />
            </SCThemeProvider>,
        )

        expect(asFragment()).toMatchSnapshot()
        expect(queryByTestId("footer-main-links-country-selector-saudi-arabia-ar")).toBeTruthy()
        expect(queryByTestId("footer-main-links-country-selector-saudi-arabia-en")).toBeTruthy()
        expect(queryByTestId("footer-main-links-country-selector-united-arab-emirates-ar")).toBeTruthy()
    })

    it("should match the snapshot template when there is no languages in CountryLink", () => {
        const customData = {
            ...props,
            elements: [
                {
                    type: "CountryLink",
                    url: "",
                    openInNewWindow: false,
                    name: "CountryLink",
                    icon: "",
                    text: "Saudi Arabia",
                    accessibilityText: "Saudi Arabia",
                    tooltip: "",
                    accessibilityTooltip: "",
                    description: "",
                    accessibilityDescription: "",
                },
            ],
        }
        const {asFragment} = render(
            <SCThemeProvider theme={mockTheme}>
                <CountrySelector {...customData} />
            </SCThemeProvider>,
        )

        expect(asFragment()).toMatchSnapshot()
    })
})
