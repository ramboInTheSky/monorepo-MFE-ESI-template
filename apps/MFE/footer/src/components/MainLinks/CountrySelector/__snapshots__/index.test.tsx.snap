// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - MainLinks/CountrySelector:  should match the snapshot template 1`] = `
<DocumentFragment>
  <li
    class="components__CountrySelectorLanguageList-sprnyc-1 jSffgl"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-space-between"
    >
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5"
          data-testid="footer-main-links-country-selector-title-saudi-arabia"
        >
          Saudi Arabia
        </h5>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <a
          aria-label="Change to Saudi Arabia in Arabic"
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover components__CountrySelectorLanguageLink-sprnyc-0 jSxFFF MuiTypography-colorPrimary"
          data-testid="footer-main-links-country-selector-saudi-arabia-ar"
          href="http://superman.com//www.amido.com/sa/ar"
          target="_self"
        >
          AR
        </a>
        <span>
          |
        </span>
        <a
          aria-label="Change to Saudi Arabia in English"
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover components__CountrySelectorLanguageLink-sprnyc-0 jSxFFF MuiTypography-colorPrimary"
          data-testid="footer-main-links-country-selector-saudi-arabia-en"
          href="http://superman.com//www.amido.com/sa/en"
          target="_self"
        >
          <span>
            <!--mp_trans_disable_start-->
            EN
            <!--mp_trans_disable_end-->
          </span>
        </a>
      </div>
    </div>
  </li>
  <li
    class="components__CountrySelectorLanguageList-sprnyc-1 jSffgl"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-space-between"
    >
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5"
          data-testid="footer-main-links-country-selector-title-united-arab-emirates"
        >
          United Arab Emirates
        </h5>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <a
          aria-label="Change to United Arab Emirates in Arabic"
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover components__CountrySelectorLanguageLink-sprnyc-0 jSxFFF MuiTypography-colorPrimary"
          data-testid="footer-main-links-country-selector-united-arab-emirates-ar"
          href="http://superman.com//www.amido.com/ae/ar"
          target="_self"
        >
          AR
        </a>
      </div>
    </div>
  </li>
</DocumentFragment>
`;

exports[`Components - MainLinks/CountrySelector:  should match the snapshot template when there is no languages in CountryLink 1`] = `<DocumentFragment />`;
